lib/wootric/index.js

```diff
@@ -0,0 +1,146 @@
+    survey_callback: this.trackSurvey
+  };
+
+  window.wootricSettings.account_token = this.options.account_token;
+  //Load our library, once it is loaded, we are ready
+  this.load('library', function(){
+    window.wootric("run");
+    self.ready();
+  });
+};
+
+// Track the survey result.  This is called back once the result is sent to Wootric
+// by way of the wootricSettings.response_callback callback
+Wootric.prototype.trackSurveyResponse = function(score, text){
+  this.track('NPS Response', {
+    email: window.wootricSettings.email,
+    createdAt: window.wootricSettings.created_at,
+    score: score,
+    text: text
+  })
+}
+
+// Track the survey result.  This is called back once the result is sent to Wootric
+// by way of the wootricSettings.decline_callback callback
+Wootric.prototype.trackSurveyDecline = function(){
+  this.track('NPS Decline', {
+    email: window.wootricSettings.email,
+    createdAt: window.wootricSettings.created_at
+  })
+}
+
+// Track the survey result.  This is called back once the result is sent to Wootric
+// by way of the wootricSettings.survey_callback callback
+Wootric.prototype.trackSurvey = function(){
+  this.track('NPS Survey', {
+    email: window.wootricSettings.email,
+    createdAt: window.wootricSettings.created_at
+  })
+}
+
+/**
+ * Has the Wootric library been loaded yet?
+ *
+ * @return {Boolean}
+ */
+
+Wootric.prototype.loaded = function(){
+  // TODO: fill in the logic required to check if the library has already been
+  //We are always ready since we are just setting a global variable in initialize
+  return window.wootric;
+};
+
+/**
+ * Identify a user.
+ *
+ * @param {Facade} identify
+ */
+
+Wootric.prototype.identify = function(identify){
+  //console.log("calling identify")
+  var traits = identify.traits();
+  var email = traits.trait.email;
+  var created_at = traits.trait.createdAt;
+  var unixDt = Date.parse(created_at) || +created_at;
+  if (unixDt) {
+    window.wootricSettings.created_at = unixDt;
+  }
+  window.wootricSettings.email = email;
+  //Set the rest of the traits as properties
+  window.wootricSettings.properties = traits.trait;
+  //remove the email and createdAt properties
+  delete window.wootricSettings.properties.email;
+  delete window.wootricSettings.properties.createdAt;
+  //Track page visit
+  this.page();
+  //console.log("wootricSettings", window.wootricSettings);
+  };
+
+//Store the last page name tracked
+Wootric.lastPageTracked = null;
+
+/**
+ * Page.
+ *
+ * @param {Page} page
+ */
+Wootric.prototype.page = function(page){
+  //console.log("calling page");
+  var opts = this.options;
+  //console.log(opts);
+  if (opts.account_token) {
+    //Only track page if we haven't already tracked it
+    if (this.lastPageTracked !== window.location) {
+      this.lastPageTracked = window.location;
+      this.createTrackingPixel(window.location, this.options.account_token, window.wootricSettings.email, window.wootricSettings.created_at);
```

Comment:
you shouldn't need a separate function here to load the tracking pixel. you can set this pixel on the prototype with a name like [here](https://github.com/segmentio/analytics.js-integrations/blob/7a212232baee1054b8a7532bb6f76cfeb854a3ec/lib/comscore/index.js#L20).. 
`.tag('page', '<script src="//d8myem934l1zi.cloudfront.net/pixel.gif?account_token={{ account_token }}&email={{ email }}...` and then load that pixel in the page function with passing the relevant parameters